### PR TITLE
feat: manage timesyncd package on Debian >= 11 and Ubuntu >= 20.04

### LINKED ISSUE #294: systemd-timesyncd package should be managed, if required
On Debian, there's a package `systemd-timesyncd` which should be managed, when `manage_timesyncd => true`.

I'll send in a PR asap

### PR DESCRIPTION
**Pull request by saz**
Fixes #294

### CODE REVIEW

**Review by ekohl — COMMENTED** [review]
I'm leaning to disabling `RSpec/RepeatedExample` to (at least locally) to make Rubocop happy. I don't mind the repetition here.

**Comment by saz** [comment]
@ekohl I've just pushed a change removing the repeated examples, although I'd say it's better to keep them, just to be sure.

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -263,6 +264,14 @@
           it { is_expected.not_to create_service('systemd-resolved').with_enable(true) }
           it { is_expected.not_to create_service('systemd-networkd').with_ensure('running') }
           it { is_expected.not_to create_service('systemd-networkd').with_enable(true) }
+
+          if facts[:os]['family'] == 'Debian'
+            if Puppet::Util::Package.versioncmp(facts[:os]['release']['full'], '20.04') >= 0 || Puppet::Util::Package.versioncmp(facts[:os]['release']['major'], '11') >= 0
```
Comment (ekohl) [inline spec/classes/init_spec.rb:269]: You could even make sure it's not set on non-Debian:
```suggestion
          if facts[:os]['family'] == 'Debian' && (Puppet::Util::Package.versioncmp(facts[:os]['release']['full'], '20.04') >= 0 || Puppet::Util::Package.versioncmp(facts[:os]['release']['major'], '11') >= 0)
```

**Review by ekohl — APPROVED** [review]

File: spec/classes/init_spec.rb (hunk 1 above)
Comment (saz) [reply spec/classes/init_spec.rb]: As my condition wasn't working, I've changed it.

**Review by ekohl — APPROVED** [review]

**Comment by bastelfreak** [comment]
@jhoblitt while I'm the person that added the enhancement label, I now think bug is better. Debian 10/Ubuntu 20.04 was already in metadata.json, so one could argue this PR doesn't add a new feature/support, it fixes existing code.

**Comment by jhoblitt** [comment]
@bastelfreak copy. Thank you.  I will reroll the changelog.

**Pull request merged**